elements/cc-rclone/bin/cc-mount-object-store

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+exit_failure() {
+  echo "$(date '+%Y-%m-%d %H:%M:%S'): Failed to create object store mount!" >> my_mounting_point.failure
+}
+
+trap 'exit_failure' EXIT
+
+set -e
+
+if [ ! -f /home/cc/openrc ]; then
+  OPENRC=$(cc-read-vendordata "openrc")
+  if [ "${OPENRC}" != "null" ]; then
+    cc-read-vendordata "openrc" >  /home/cc/openrc
+    sed -i 's/^ *//g' /home/cc/openrc
+  else
+    echo "$(date '+%Y-%m-%d %H:%M:%S'): Unable to read OPENRC" >> /home/cc/my_mounting_point.failure
+    exit 1
+  fi
+fi
+
+source /home/cc/openrc
+if [ -z "${OS_STORAGE_URL}" ]; then
+  echo "$(date '+%Y-%m-%d %H:%M:%S'): OS_STORAGE_URL was not set" >> /home/cc/my_mounting_point.failure
+  REGION=$(cc-read-vendordata "region")
+  if [ "${REGION}" = "KVM@TACC" ]; then
+    echo "$(date '+%Y-%m-%d %H:%M:%S'): KVM@TACC does not have an object store available" >> /home/cc/my_mounting_point.failure
+  fi
+  exit 1
+fi
+
+mkdir /home/cc/my_mounting_point
+touch /home/cc/my_mounting_point/THIS_IS_NOT_MOUNTED
+mkdir -p /home/cc/.config/rclone/
+
+EC2_STORAGE_ENDPOINT=${OS_STORAGE_URL%/swift*}
+EC2_CREDS="$(openstack ec2 credential create -f json)"
```

Comment:
Main concern here is this will  create a new credential every time it runs

Reply:
Are there possibilities for generating a single long-term use credentials up front but reusing it for subsequent mounts of this?

Otherwise we can go the route of "user brings / manages their own credential."

Reply:
it's a little messy with the ec2 credentials. We could do something like "list credentials, and create if missing", the downside is that there's no way to tag or otherwise label ec2 credentials via the openstack api, and I'm unsure if there's any issue from reusing one that was previously created. (just dunno all the edges)

Reply:
Even with a newer rclone I still am having issues with the Swift backend and files >100mb. Maybe I'm missing something else though.

Reply:
nah, that's fair. I think just using the first entry from `openstack ec2 credential list` is reasonable, and creating it if missing.

[One thing I wasn't clear on from our current cloudfuse docs](https://chameleoncloud.readthedocs.io/en/latest/technical/swift.html#mounting-object-store-as-a-file-system), is how the bucket gets named, and what "site" the bucket is at. I think we assume it's always named after the username, and at the "local" site, but what if the username has characters that aren't permitted in a bucket?

Reply:
To start I was just mounting the entire object store they have permissions for, but scoping it to something more specific makes sense. Instead of their username, we could also make them specify the bucket when they start the mount? It introduces friction which might reduce the willingness to use it. On the other hand, it's not _that_ much friction and helps make it more clear what/where it is mounting since they have to take some ownership over understanding and providing that argument.

Reply:
Actually, I think I might have been imagining the "mount a container named after your username" behavior, pretty sure what cloudfuse did was mount the "root" path, same as you're doing

my main concern is in classroom cases, having 200 students deleting each others files  with an rm -rf

If they've gotta run it from the commandline manyally anyway, having to specify a bucket name seems sane to me

Reply:
For automation (later) , something leveraging systemd mount units could work nicely, since it keeps the user from having to worry about their process dying, or racing the network on startup
https://rclone.org/commands/rclone_mount/#systemd